### PR TITLE
docs: added guide on using HTTP2

### PR DESCRIPTION
**Pull request by guptadeepak8**
fix:- tailcallhq/tailcall#810

### CODE REVIEW

**Comment by guptadeepak8** [comment]
@tusharmath Apologies for the inconvenience caused by PR #128. I would greatly appreciate it if the merge could be done without bounty. Thank you.

File: docs/guides/http2.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+description: Enhance web performance and secure communication by leveraging HTTP/2 with Tailcall.
+---
+
+HTTP/2 is a modern protocol that enhances improves web performance by introducing features like multiplexing, header compression, and more efficient connection handling.
+
+Tailcall seamlessly supports HTTP/2 on both the server (**ingress**) and client (**egress**) sides. This means that Tailcall facilitates the utilization of HTTP/2 protocol for both incoming requests to your server and outgoing requests from your server to other services.
+
+### Ingress Side Configuration
+
+Tailcall's `@http` operator facilitates integration with HTTP/2 for server-side configurations. By specifying HTTP2 in the http field of the schema definition, you enable the utilization of the HTTP/2 protocol:
```
Comment (tusharmath) [inline docs/guides/http2.md]:
```suggestion
Tailcall's `@server` operator facilitates integration with HTTP/2 for server-side configurations. By specifying HTTP2 in the http field of the schema definition, you enable the utilization of the HTTP/2 protocol:
```

File: docs/guides/http2.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+```graphql showLineNumbers
+schema
+  @server(
+    port: 8000
+    graphiql: true
+    version: HTTP2
+    cert: "./cert.pem"
+    key: "./key.pem"
```
Comment (tusharmath) [inline docs/guides/http2.md]:
```suggestion
    cert: "./cert.pem"
    key: "./key.pem"
```

Use `@link` instead. These settings don't exist anymore.

File: docs/guides/http2.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+3. **Generate certificate signing request(CSR)**
+
+```bash
+openssl req -new -key key.pem -out csr.pem
+```
+
+Running this command prompts you with questions to provide information for the certificate.
+
+![http2-csr.png](../../static/images/docs/http2-csr.png)
```
Comment (tusharmath) [inline docs/guides/http2.md]: can we use https://carbon.now to generate these screenshots instead?

File: docs/guides/http2.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+5. **Clean up intermediate files**
+
+```bash
+rm csr.pem
+```
+
+Once the certificate signing request (CSR) has been used to generate the self-signed certificate (cert.pem), the CSR file (csr.pem) is no longer needed this command is done to clean up intermediate files generated during the certificate creation process.
+
+By following these steps, you can create the necessary certificates locally for use with Tailcall's HTTP/2 configuration, ensuring secure communication between your server and clients.
```
Comment (tusharmath) [inline docs/guides/http2.md]: Let's add a screenshot of using HTTP2, perhaps e a screenshot of curl upgrading HTTP2, or chrome devtools?

File: docs/guides/http2.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+## Egress Side Implementation
+
+On the egress side, Tailcall's @http operator provides granular control over outgoing requests, enabling efficient utilization of HTTP/2 features:
+
```
Comment (tusharmath) [inline docs/guides/http2.md:81]:
```suggestion
On the egress side, Tailcall's `@http` operator provides granular control over outgoing requests, enabling efficient utilization of HTTP/2 features:
```

File: docs/guides/http2.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
+---
+title: Leveraging HTTP/2
+description: Enhance web performance and secure communication by leveraging HTTP/2 with Tailcall.
+---
+
+HTTP/2 is a modern protocol that enhances improves web performance by introducing features like multiplexing, header compression, and more efficient connection handling.
+
+Tailcall seamlessly supports HTTP/2 on both the server (**ingress**) and client (**egress**) sides. This means that Tailcall facilitates the utilization of HTTP/2 protocol for both incoming requests to your server and outgoing requests from your server to other services.
```
Comment (tusharmath) [inline docs/guides/http2.md]: Perhaps a more concise version could be — 
Tailcall supports HTTP/2 for both server (ingress) and client (egress) operations, enabling the protocol for incoming and outgoing server requests.

File: docs/guides/http2.md (hunk 4 above)
Comment (guptadeepak8) [reply docs/guides/http2.md]: Screenshot (675)

@tusharmath  Is this screenshot correct for showing protocol h2

Comment (guptadeepak8) [reply docs/guides/http2.md]: or should i add the comparison between protocols
Screenshot (679)


Screenshot (680)

File: docs/guides/http2.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+4. **Generate Self-Signed Certificate**
+
+   ```bash
+   openssl x509 -req -days 365 -in csr.pem -signkey key.pem -out cert.pem
+   ```
+
+   This generates a self-signed certificate valid for 365 days using the CSR from step 3 and the private key from step 2. The validity period can be adjusted by changing the number of days (-days). A "Signature ok" prompt confirms the successful creation.
+
+   ![http/2-cert.png](../../static/images/docs/http2-cert.png)
```
Comment (tusharmath) [inline docs/guides/http2.md:42]: can you please update the image such that the BG is transparent? Shadows are fine but without the grey background.

**Comment by tusharmath** [comment]
The images aren't web-optimized, can do two things — 

1. Improve resolution, they look pixelated on my Mac.
2. Optimize them for web, one of them is 219kb, which is quite a lot.

**Comment by guptadeepak8** [comment]
@tusharmath  update the images

**Review by tusharmath — APPROVED** [review]

**Pull request merged**